lib/c-api/src/wasm_c_api/externals/function.rs

```diff
@@ -159,23 +87,18 @@ pub unsafe extern "C" fn wasm_func_new_with_env(
             .take()
             .into_iter()
             .map(TryInto::try_into)
-            .collect::<Result<Vec<Val>, _>>()
+            .collect::<Result<Vec<Value>, _>>()
             .expect("Result conversion failed");
 
         Ok(processed_results)
     };
+    let mut lck = ctx.lock().unwrap();
```

Comment:
We shouldn't lock the context. The Wasm-C-API rely on soundness for using their API. Meaning: it puts the responsibility on the user to use safely on different threads, not on the API itself.